### PR TITLE
[MODORDERS-730] Item not appearing under holdings after instance connection change

### PR DESCRIPTION
**Pull request by Abdulkhakimov**
## Purpose
Provide to make operations with piece holdings.

## Approach
Retrieving holdings by poLineId from orders storage piece endpoint and execute holding ids from connected pieces.

## Learning
[MODORDERS-730](https://issues.folio.org/browse/MODORDERS-730)

## Pre-Merge Checklist:
Before merging this PR, please go through the following list and take appropriate actions.

- Does this PR meet or exceed the expected quality standards?
  - [x] Code coverage on new code is 80% or greater
  - [x] Duplications on new code is 3% or less
  - [x] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [x] Were any API paths or methods changed, added or removed?
  - [ ] Were there any schema changes?
  - [ ] Did any of the interface versions change?
  - [x] Were permissions changed, added, or removed?
  - [ ] Are there new interface dependencies?
  - [ ] There are no breaking changes in this PR.
  
If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?  
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.  

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

File: src/main/java/org/folio/service/orders/lines/update/instance/WithHoldingOrderLineUpdateInstanceStrategy.java [hunk 1]
```diff
@@ -62,62 +71,66 @@ protected CompletableFuture<Void> processHoldings(OrderLineUpdateInstanceHolder
...
-      .map(Location::getHoldingId)
-      .filter(Objects::nonNull)
-      .collect(toList());
-    holdingIds.forEach(id -> holder.addHoldingRefsToStoragePatchOrderLineRequest(id, id));
-
-    return inventoryManager.getHoldingsByIds(holdingIds, requestContext)
-        .thenCompose(holdings -> inventoryManager.updateInstanceForHoldingRecords(holdings, newInstanceId, requestContext));
+    return pieceStorageService.getPiecesByPoLineId(PoLineCommonUtil.convertToCompositePoLine(holder.getStoragePoLine()), requestContext)
+      .thenApply(pieces -> {
+        List<String> holdingIds = Stream.concat(pieces.stream().map(Piece::getHoldingId),
```
Comment (okolawole-ebsco) [inline src/main/java/org/folio/service/orders/lines/update/instance/WithHoldingOrderLineUpdateInstanceStrategy.java:76]: please add a small comment describing what the code in the `thenApply` is doing.

Comment (Abdulkhakimov) [reply src/main/java/org/folio/service/orders/lines/update/instance/WithHoldingOrderLineUpdateInstanceStrategy.java]: Done

File: src/main/java/org/folio/service/orders/lines/update/instance/WithHoldingOrderLineUpdateInstanceStrategy.java [hunk 2]
```diff
@@ -62,62 +71,66 @@ protected CompletableFuture<Void> processHoldings(OrderLineUpdateInstanceHolder
...
+        .thenCompose(newHoldingId -> {
+          holder.addHoldingRefsToStoragePatchOrderLineRequest(holdingId, newHoldingId);
+          CompositePoLine compositePoLine = PoLineCommonUtil.convertToCompositePoLine(holder.getStoragePoLine());
+          return updateItemsHolding(holdingId, newHoldingId, compositePoLine.getId(), requestContext);
+        }));
+    }).thenCompose(v -> CompletableFuture.allOf(futures.toArray(new CompletableFuture[0])));
+  }
 
-    return CompletableFuture.allOf(futures.toArray(new CompletableFuture[0]));
+  private CompletableFuture<Void> prepareHoldings(OrderLineUpdateInstanceHolder holder, RequestContext requestContext, Consumer<Location> consumer) {
```
Comment (okolawole-ebsco) [inline src/main/java/org/folio/service/orders/lines/update/instance/WithHoldingOrderLineUpdateInstanceStrategy.java:120]: can you add a comment here describing how holdings are prepared?

File: src/main/java/org/folio/service/orders/lines/update/instance/WithHoldingOrderLineUpdateInstanceStrategy.java [hunk 3]
```diff
@@ -62,62 +71,66 @@ protected CompletableFuture<Void> processHoldings(OrderLineUpdateInstanceHolder
...
-      .stream()
-      .map(Location::getHoldingId)
-      .filter(Objects::nonNull)
-      .collect(toList());
-    holdingIds.forEach(id -> holder.addHoldingRefsToStoragePatchOrderLineRequest(id, id));
-
-    return inventoryManager.getHoldingsByIds(holdingIds, requestContext)
-        .thenCompose(holdings -> inventoryManager.updateInstanceForHoldingRecords(holdings, newInstanceId, requestContext));
+    return pieceStorageService.getPiecesByPoLineId(PoLineCommonUtil.convertToCompositePoLine(holder.getStoragePoLine()), requestContext)
+      // Extraction unique holdingIds from pieces and POL
```
Comment (okolawole-ebsco) [inline src/main/java/org/folio/service/orders/lines/update/instance/WithHoldingOrderLineUpdateInstanceStrategy.java:75]: nitpick: think you mean "extract" not "extraction"

Comment (Abdulkhakimov) [reply src/main/java/org/folio/service/orders/lines/update/instance/WithHoldingOrderLineUpdateInstanceStrategy.java]: I decided to extract as new method following by best practices, are you ok?

File: src/main/java/org/folio/service/orders/lines/update/instance/WithHoldingOrderLineUpdateInstanceStrategy.java (hunk 2 above)
Comment (Abdulkhakimov) [reply src/main/java/org/folio/service/orders/lines/update/instance/WithHoldingOrderLineUpdateInstanceStrategy.java]: Renamed the method which definitely describe main purpose.

File: src/main/java/org/folio/service/orders/lines/update/instance/WithHoldingOrderLineUpdateInstanceStrategy.java (hunk 3 above)
Comment (okolawole-ebsco) [reply src/main/java/org/folio/service/orders/lines/update/instance/WithHoldingOrderLineUpdateInstanceStrategy.java]: I was only referring to your grammar in the comment. I am good with putting things in a new method.

**Review by okolawole-ebsco — APPROVED** [review]

**Review by SerhiiNosko — APPROVED** [review]

**Pull request merged**